src/main/scala/temple/generate/target/openapi/OpenAPIBuilder.scala

```diff
@@ -0,0 +1,148 @@
+package temple.generate.target.openapi
+
+import io.circe.syntax._
+import io.circe.yaml.Printer
+import temple.DSL.semantics.AttributeType._
+import temple.DSL.semantics.{Annotation, Attribute}
+import temple.collection.FlagMapView
+import temple.generate.Endpoint._
+import temple.generate.target.openapi.OpenAPIFile.{Components, Info}
+import temple.generate.target.openapi.OpenAPIType._
+
+import scala.collection.immutable.ListMap
+import scala.collection.mutable
+
+private class OpenAPIBuilder private (name: String, version: String, description: String = "") {
+
+  private val errorTracker = FlagMapView(
+    400 -> OpenAPIBuilder.generateError("Invalid request", "Invalid request parameters: name"),
+    404 -> OpenAPIBuilder.generateError("ID not found", "Object not found with ID 1"),
+    500 -> OpenAPIBuilder.generateError(
+      "The server encountered an error while serving this request",
+      "Unable to reach user service: connection timeout",
+    ),
+  )
+
+  private val paths = mutable.Map[String, mutable.Map[HTTPVerb, Path]]()
+
+  private def path(url: String): mutable.Map[HTTPVerb, Path] =
+    paths.getOrElseUpdate(url, mutable.Map())
+
+  private def isServerAttribute(attribute: Attribute): Boolean = attribute.accessAnnotation contains Annotation.Server
+
+  private def isClientAttribute(attribute: Attribute): Boolean =
+    attribute.accessAnnotation.isEmpty || (attribute.accessAnnotation contains Annotation.Client)
+
+  private def attributeToOpenAPIType(attribute: Attribute): OpenAPISimpleType = attribute.attributeType match {
+    case BoolType     => OpenAPISimpleType("boolean")
+    case DateType     => OpenAPISimpleType("string", "date")
+    case DateTimeType => OpenAPISimpleType("string", "date-time")
+    case TimeType     => OpenAPISimpleType("string", "time")
+    case BlobType(size) =>
+      val maxLength = size.map("maxLength" -> _.asJson)
+      OpenAPISimpleType("string", Seq(maxLength).flatten: _*)
+    case StringType(max, min) =>
+      val minLength = min.map("minLength" -> _.asJson)
+      val maxLength = max.map("maxLength" -> _.asJson)
+      OpenAPISimpleType("string", Seq(minLength, maxLength).flatten: _*)
+    case IntType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "int64" else "int32", Seq(minimum, maximum).flatten: _*)
+    case FloatType(max, min, precision) =>
+      val minimum = min.map("minimum" -> _.asJson)
+      val maximum = max.map("maximum" -> _.asJson)
+      OpenAPISimpleType("number", if (precision > 4) "double" else "float", Seq(minimum, maximum).flatten: _*)
```

Comment:
A little repetitive, but not sure how to avoid it. `Seq().flatten: _*` is the simplest way I could find to optionally supply arguments to a variadic function